core/src/main/scala/kafka/controller/ElectionUtil.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.controller
+
+import kafka.api.LeaderAndIsr
+import org.apache.kafka.common.TopicPartition
+
+case class ElectionResult(topicPartition: TopicPartition, leaderAndIsr: Option[LeaderAndIsr], liveReplicas: Seq[Int])
+
+object ElectionUtil {
```

Comment:
Why not call it `ElectionLeader` and then remove the `leader` prefix from the method names?

Reply:
Or something without the `Util`. In Scala, there's rarely a reason to use `Util` in the name of classes/objects.

Reply:
I chose `Util` just because it made the expected usage clear. `ElectionLeader` seems a bit unnatural. Maybe just `Election`?

Reply:
Yeah, Election was the first thing that came to mind.